pkg/state/state_cni_plugins_test.go

```diff
@@ -0,0 +1,346 @@
+		Expect(err).NotTo(HaveOccurred())
+		Expect(status).To(BeEquivalentTo(state.SyncStateNotReady))
+		By("Verify DaemonSet")
+		ds := &appsv1.DaemonSet{}
+		err = client.Get(context.Background(), types.NamespacedName{Namespace: namespace, Name: "cni-plugins-ds"}, ds)
+		Expect(err).NotTo(HaveOccurred())
+		expectedDs := getExpectedMinimalCniPluginDS()
+		expectedDs.Spec.Template.Spec.Volumes[0].VolumeSource.HostPath.Path = "/var/lib/cni/bin"
+		By("Verify CNI bin dir")
+		Expect(ds.Spec).To(BeEquivalentTo(expectedDs.Spec))
+	})
+
+	It("should render Daemonset with Resources when specified in CR", func() {
+		By("Sync")
+		cr := getMinimalNicClusterPolicyWithCNIPlugins()
+		cpu, _ := resource.ParseQuantity("1")
+		mem, _ := resource.ParseQuantity("1Gi")
+		cr.Spec.SecondaryNetwork.CniPlugins.ContainerResources = []mellanoxv1alpha1.ResourceRequirements{
+			{
+				Name: "cni-plugins",
+				Requests: v1.ResourceList{
+					v1.ResourceCPU:    cpu,
+					v1.ResourceMemory: mem,
+				},
+				Limits: v1.ResourceList{
+					v1.ResourceCPU:    cpu,
+					v1.ResourceMemory: mem,
+				},
+			},
+		}
+		_, err := cniPluginsState.Sync(context.Background(), cr, catalog)
+		Expect(err).NotTo(HaveOccurred())
+		By("Verify DaemonSet")
+		ds := &appsv1.DaemonSet{}
+		err = client.Get(context.Background(), types.NamespacedName{Namespace: namespace, Name: "cni-plugins-ds"}, ds)
+		Expect(err).NotTo(HaveOccurred())
+		rq := v1.ResourceRequirements{
+			Requests: v1.ResourceList{
+				v1.ResourceCPU:    cpu,
+				v1.ResourceMemory: mem,
+			},
+			Limits: v1.ResourceList{
+				v1.ResourceCPU:    cpu,
+				v1.ResourceMemory: mem,
+			},
+		}
+		expectedDs := getExpectedMinimalCniPluginDS()
+		expectedDs.Spec.Template.Spec.Containers[0].Resources = rq
+		By("Verify container resources")
+		Expect(ds.Spec).To(BeEquivalentTo(expectedDs.Spec))
+	})
+
+	It("should create Daemonset and delete if Spec is nil", func() {
+		By("Sync")
+		cr := getMinimalNicClusterPolicyWithCNIPlugins()
+		status, err := cniPluginsState.Sync(context.Background(), cr, catalog)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(status).To(BeEquivalentTo(state.SyncStateNotReady))
+		By("Verify DaemonSet")
+		ds := &appsv1.DaemonSet{}
+		err = client.Get(context.Background(), types.NamespacedName{Namespace: namespace, Name: "cni-plugins-ds"}, ds)
+		Expect(err).NotTo(HaveOccurred())
+		expectedDs := getExpectedMinimalCniPluginDS()
+		Expect(ds.Spec).To(BeEquivalentTo(expectedDs.Spec))
+		By("Set spec to nil and Sync")
+		cr.Spec.SecondaryNetwork.CniPlugins = nil
+		status, err = cniPluginsState.Sync(context.Background(), cr, catalog)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(status).To(BeEquivalentTo(state.SyncStateNotReady))
+		By("Verify DaemonSet is deleted")
+		ds = &appsv1.DaemonSet{}
+		err = client.Get(context.Background(), types.NamespacedName{Namespace: namespace, Name: "cni-plugins-ds"}, ds)
+		Expect(errors.IsNotFound(err)).To(BeTrue())
+	})
```

Comment:
nit: maybe add 2 contexts, one for the rendered manifests and one for the Sync flow without what is rendered.

Reply:
Not sure which test should go where, all of them are using `Sync`

Reply:
Basically, what I was thinking is:
* Add context for tests checking the flow of Sync (i.e. error handling, correct garbage collection etc)
  * https://github.com/Mellanox/network-operator/blob/30d9595d02cd033e57543f9f20bae02f4d03023f/pkg/state/state_cni_plugins.go#L93-L95
  * https://github.com/Mellanox/network-operator/blob/30d9595d02cd033e57543f9f20bae02f4d03023f/pkg/state/state_cni_plugins.go#L98-L100
  * https://github.com/Mellanox/network-operator/blob/30d9595d02cd033e57543f9f20bae02f4d03023f/pkg/state/state_cni_plugins.go#L112-L114
  * etc
* Add context for tests checking the manifests rendered by Sync
  * Did it render the image pull secret correctly?
  * Did it render container resources correctly?
  * etc

Reply:
Done